src/components/ChatContent/Messages/BlockMenu/ReportDialog.vue

```diff
@@ -0,0 +1,300 @@
+      <div style="display: inline-block; margin: 0px 12px;">
+        <input type="checkbox" v-model="ignore">
+        <label for="report-ignore">Ignore User</label>
+      </div>
+      <div style="display: inline-block; margin: 0px 12px;">
+        <input type="checkbox" v-model="report">
+        <label for="report-report">Report User</label>
+      </div>
+    </div>
+    <div style="margin-top: 15px;" v-show="report">
+      <label for="report-message">Additional Comments:</label>
+      <textarea v-model="reportComments" style="resize: none; height: 60px;"></textarea>
+    </div>
+    <div style="margin-top: 15px;">
+      <div
+        class="blue-button blue-button-interactive"
+        style="width: 40%; display: inline-block; vertical-align: middle;"
+        @click="closeModal"
+      >Cancel</div>
+      <div
+        class="green-button green-button-interactive"
+        style="width: 40%; display: inline-block; float: right; vertical-align: middle;"
+        @click="submit"
+      >Continue</div>
+    </div>
+  </modal>
+</template>
+
+<script lang="ts">
+  import { Component, Prop } from 'vue-property-decorator';
+  import Vue from '@/types/vue';
+  import { Client } from 'irc-framework';
+  import parseUsername from '@/tools/parseUsername';
+  import User from '../../../../types/user';
+  import Message from '../../../../types/message';
+
+@Component({})
+  export default class ReportDialog extends Vue {
+  report = false;
+
+  ignore = false;
+
+  message: Message | null = null;
+
+  get userToReport(): User {
+    return this.message ? this.message.user : User.annonymous;
+  }
+
+  $refs!: {
+    reportModal: HTMLFormElement;
+  };
+
+  reportComments: string = '';
+
+  submit() {
+    if (this.ignore) {
+      this.$store.commit('ignore', this.userToReport.username);
+    }
+    const client = this.$store.state.client!;
+    if (this.report) {
+      client.say(
+        '#ops-notifications',
+        `[REPORT] Reporting ${this.userToReport.username} (${
+          this.userToReport.uid
+        }) by ${this.$store.state.currentUser.username} (${
+          this.$store.state.currentUser.uid
+        }).\r\n`,
+      );
+      if (this.message) {
+        client.say(
+          '#ops-notifications',
+          `[REPORTED MESSAGE] ${this.message!.message}\r\n`,
+        );
+      }
+      client.say(
+        '#ops-notifications',
+        `[REPORT REASON] ${this.reportComments}\r\n`,
+      );
+    }
+    this.closeModal();
+  }
+
+  closeModal() {
+    this.$modal.hide('reportModal');
+    this.reportComments = '';
+  }
+
+  created() {
+    this.$store.subscribe((mutation, state) => {
+      if (mutation.type === 'openReportModal') {
+        this.message = mutation.payload.message;
+        this.report = mutation.payload.defaults.report;
+        this.ignore = mutation.payload.defaults.ignore;
+        this.$modal.show('reportModal');
+      }
+    });
+  }
+  }
+</script>
+
+<style lang="scss">
```

Comment:
The element with `.report-modal` isn't in this file, but rather a child of the modal component, so if I scope its css it won't reach it.
```  <modal
    name="reportModal"
    classes="report-dialog"
    :max-width="350"
    height="auto"
    :adaptive="true"
  >```

Reply:
Well, first off it should be `class` not `classes`, which should still be causing issues? That being said, I'm pretty sure scoped css does apply to children. Barring that:
> With scoped, the parent component's styles will not leak into child components. However, a child component's root node will be affected by both the parent's scoped CSS and the child's scoped CSS. This is by design so that the parent can style the child root element for layout purposes.
https://vue-loader.vuejs.org/guide/scoped-css.html#child-component-root-elements
(There's also syntax for modifying things deeper in child components!)

Reply:
The `modal` component contains the darkened background as well, and so the actual modal (which is what gets the class) isn't the root element.
And it is classes for that reason - it's a prop used by the `modal` component.

Reply:
You can use deep selectors for this purpose https://vue-loader.vuejs.org/guide/scoped-css.html#deep-selectors